git_machete/utils.py

```diff
@@ -94,11 +94,27 @@ def debug(msg: Optional[str] = None) -> None:
     if debug_mode:
         function_name = bold(inspect.stack()[1].function)
         args, _, _, values = inspect.getargvalues(inspect.stack()[1].frame)
-        function_args = bold(f"({', '.join([arg+'='+str(values[arg]) if arg != 'self' else 'self' for arg in args])})")
+
+        args_to_be_redacted_by_name = {'access_token', 'password', 'secret', 'token'}
+        for arg in args_to_be_redacted_by_name:
+            if arg in values.keys():
+                values[arg] = '***'
+
+        args_to_be_redacted_by_value = ['ghp_', 'gho_', 'ghu_', 'ghs_', 'ghr_']  # https://github.blog/2021-04-05-behind-githubs-new-authentication-token-formats/
+        for arg, value in values.items():
+            if any(arg_ in str(value) for arg_ in args_to_be_redacted_by_value):
+                values[arg] = '***'
```

Comment:
Or maybe even:
```suggestion
        args_to_be_redacted_by_name = {'access_token', 'password', 'secret', 'token'}
        args_to_be_redacted_by_value = ['ghp_', 'gho_', 'ghu_', 'ghs_', 'ghr_']  # https://github.blog/2021-04-05-behind-githubs-new-authentication-token-formats/

        for arg, value in values.items():
            if arg in args_to_be_redacted_by_name or any(arg_ in str(value) for arg_ in args_to_be_redacted_by_value):
                values[arg] = '***'
```

AFAICS that's equivalent, and easier to think of (just one loop, instead of two over two different collections)

Reply:
Also, `arg_` in `for`-comprehension shouldn't be called this way, rather `value_`

Reply:
`args_to_be_redacted_by_value` should also rather be called `values_to_be_redacted` 🤔

Reply:
Hmm, i specifically separated them to not get confused between the two of them in one big loop, thought it would be easier/cleaner to read/understand, @PawelLipski wdyt? :D

Reply:
> Hmm, i specifically separated them to not get confused between the two of them in one big loop

Okay, no strong conviction here, leaving to your judgement